scripts/add_electricity.py

```diff
@@ -470,12 +470,26 @@ def attach_hydro(n, costs, ppl):
 
     _add_missing_carriers_from_costs(n, costs, carriers)
 
+    if snakemake.params.alternative_clustering:
+        # bus to region mapping for hydro powerplants
+        bus_to_region_id = (
+            ppl.drop_duplicates(subset="bus").set_index("bus")["region_id"].to_dict()
+        )
+
     ppl = (
         ppl.query('carrier == "hydro"')
         .reset_index(drop=True)
         .rename(index=lambda s: str(s) + " hydro")
     )
 
+    with xr.open_dataarray(snakemake.input.profile_hydro) as inflow:
+        if snakemake.params.alternative_clustering:
+            for bus in inflow.indexes["plant"]:
+                if bus in bus_to_region_id:
+                    region_to_update = bus_to_region_id[bus]
+                    # Update all rows in the DataFrame where region_id matches
+                    ppl.loc[ppl["region_id"] == region_to_update, "bus"] = bus
+
```

Comment:
```suggestion
    ppl = (
        ppl.query('carrier == "hydro"')
        .reset_index(drop=True)
        .rename(index=lambda s: str(s) + " hydro")
    )
    
    if snakemake.params.alternative_clustering:
        # bus to region mapping for hydro powerplants
        bus_to_region_id = (
            ppl.drop_duplicates(subset="bus").set_index("bus")["region_id"].to_dict()
        )
        with xr.open_dataarray(snakemake.input.profile_hydro) as inflow:
            for bus in inflow.indexes["plant"]:
                if bus in bus_to_region_id:
                    region_to_update = bus_to_region_id[bus]
                    # Update all rows in the DataFrame where region_id matches
                    ppl.loc[ppl["region_id"] == region_to_update, "bus"] = bus

```

Just a minor restyling; could you try this? [we can avoid 1 if clause]